### DISCUSSION TITLE
SvelteKit with Vecerl Analytics

### DISCUSSION THREAD

**Discussion by DogFoxX** [vercel/community#1041] [Help] [answered]
I just have a question on the following snippet in `+layout.svelte` from the boilerplate:

``` js
<script>
    import Header from '$lib/header/Header.svelte';
    import { webVitals } from '$lib/vitals';
    import { browser } from '$app/env';
    import { page } from '$app/stores';
    import '../app.css';

    let analyticsId = import.meta.env.VERCEL_ANALYTICS_ID;

    $: if (browser && analyticsId) {
        webVitals({
            path: $page.url.pathname,
            params: $page.params,
            analyticsId
        })
    }
</script>
```

Since it's running client-side, is it safe to expose the `VERCEL_ANALYTICS_ID` here? Is there a way to run it inside `+layout.server.js` `load()` function instead?

**Comment by amyegan** [comment, marked as answer]
Good question @DogFoxX! The ID helps the [Web Vitals API](https://vercel.com/docs/concepts/analytics/api#) know which project owns the data so it can send it to the correct dashboard. It does not grant anyone access to your project settings or code. You should be able to move the ID out of the layout and into your lib helper function if you prefer.

**Comment by DogFoxX** [comment]
Thanx. I've modified the code like this (although it might not change much):

``` js
// +layout.svelte

<script>
    const { VERCEL_ANALYTICS_ID } = data;

    $: if (browser && VERCEL_ANALYTICS_ID) {
        webVitals({
            path: $page.url.pathname,
            params: $page.params,
            VERCEL_ANALYTICS_ID
        });
    };
</script>
```
``` js
// +layout.server.js

import { env } from "$env/dynamic/private";

export async function load() {

    return {
        VERCEL_ANALYTICS_ID: env.VERCEL_ANALYTICS_ID
    };
}
```

Also, please advise if I should use `import.meta.env.VERCEL_ANALYTICS_ID` rather than `env.VERCEL_ANALYTICS_ID` using `import { env } from "$env/dynamic/private"`
